src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -163,6 +166,23 @@ private static String readVersionFile(String sdkHomePath) {
     return null;
   }
 
+  public static boolean isFlutterModule(@Nullable Module module) {
+    return module != null && ModuleType.is(module, FlutterModuleType.getInstance());
+  }
+
+  public static boolean hasFlutterModule(@NotNull Project project) {
+    return ModuleUtil.hasModulesOfType(project, FlutterModuleType.getInstance());
+  }
+
+  /**
+   * Checks the workspace for any open Flutter projects.
+   *
+   * @return true if an open Flutter project is found
+   */
+  public static boolean isFluttering() {
+    return Arrays.stream(ProjectManager.getInstance().getOpenProjects()).anyMatch((Project p) -> FlutterSdkUtil.hasFlutterModule(p));
```

Comment:
Interesting.  I shied away from `e.getProject()` because sometimes it returns null but then I now notice that it's getting called more than once...  So maybe it's not null when it needs to be?